### PR TITLE
Serialize requests sent between servers

### PR DESCRIPTION
**Pull request by LukasJoswiak**

### CODE REVIEW

File: include/Client.hpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  void Start(tcp::resolver::iterator endpoint_iter);
+  void Stop();
+ private:
+  void StartConnect(tcp::resolver::iterator endpoint_iter);
+  void HandleConnect(const boost::system::error_code& ec,
+    tcp::resolver::iterator endpoint_iter);
+
+  void StartRead();
+  void HandleRead(const boost::system::error_code& ec);
```
Comment (TheJMaster) [inline include/Client.hpp]: Pls follow Google style guide: https://google.github.io/styleguide/cppguide.html
Should be `const boost::system::error_code &ec` and same for pointers.

File: include/Message.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  Message();
+  Message(std::string m, MessageType type);
+
+  std::string GetMessage() {
+    return m_;
+  }
+ private:
+  friend class boost::serialization::access;
+
+  template<class Archive> void serialize(Archive& ar,
```
Comment (TheJMaster) [inline include/Message.hpp]: Please you pascale case.

File: src/client/Client.cpp [hunk 3]
```diff
@@ -0,0 +1,266 @@
+// Copyright 2019 Justin Johnson, Lukas Joswiak, and Jack Khuu
+
+#include <boost/bind.hpp>
+
+#include "Client.hpp"
+#include "Message.hpp"
+
+std::string encode() {
```
Comment (TheJMaster) [inline src/client/Client.cpp:8]: Pls use Pascale case. Or move it.

File: src/client/Client.cpp [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    socket_.close();
+
+    // Try next available endpoint.
+    StartConnect(++endpoint_iter);
+  } else {
+    // Connection successfully established.
+    std::cout << "Connected to " << endpoint_iter->endpoint() << std::endl;
+
+    StartRead();
+
```
Comment (TheJMaster) [inline src/client/Client.cpp]: nit: remove newline

File: src/client/Client.cpp [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+    if (!line.empty()) {
+      std::cout << "Received: " << line << std::endl;
+    }
+
+    StartRead();
+  } else {
+    std::cout << "Error on read: " << ec.message() << std::endl;
+
+    // Stop();
```
Comment (TheJMaster) [inline src/client/Client.cpp:115]: ???

File: src/server/TCPConnection.cpp [hunk 6]
```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 Lukas Joswiak, Justin Johnson, Jack Khuu.
+
+#include <iostream>
+
+#include "Message.hpp"
+#include "TCPConnection.hpp"
+
+Message decode(std::string data) {
```
Comment (TheJMaster) [inline src/server/TCPConnection.cpp:8]: Pls use Pascale case.

**Review by TheJMaster — CHANGES_REQUESTED** [review]

**Comment by TheJMaster** [comment]
Pls resolve merge conflicts.

File: include/Message.hpp (hunk 2 above)
Comment (LukasJoswiak) [reply include/Message.hpp]: Cannot use Pascal case here because serialize is an overloaded method.

File: src/client/Client.cpp (hunk 3 above)
Comment (LukasJoswiak) [reply src/client/Client.cpp:8]: Will move to util class in next PR.

File: src/client/Client.cpp (hunk 5 above)
Comment (LukasJoswiak) [reply src/client/Client.cpp:115]: Added TODO.

File: src/server/TCPConnection.cpp (hunk 6 above)
Comment (LukasJoswiak) [reply src/server/TCPConnection.cpp:8]: Will update when moving to util class.

**Review by TheJMaster — APPROVED** [review]

**Pull request merged**